lib/wallet/src/Cardano/Wallet/Write/Tx/Balance.hs

```diff
@@ -892,13 +898,19 @@ balanceTransactionWithSelectionStrategyAndNoZeroAdaAdjustment
         -> Either (SelectionError WalletSelectionContext) Selection
     selectAssets' era outs utxoSelection balance fee0 seed =
         let
-            txPlutusScriptExecutionCost = maxScriptExecutionCost pp redeemers
+            txPlutusScriptExecutionCost = W.toWallet @W.Coin $
+                if null redeemers then
+                    mempty
+                else
+                    maxScriptExecutionCost (recentEra @era) pp
```

Comment:
A monoid homomorphism in the wild! (Always maps `mempty` in the source monoid to `mempty` in the target monoid.)